salt/beacons/cert_info.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+'''
+Beacon to monitor certificate expiration dates from files on the filesystem.
+
+.. versionadded:: Sodium
+
+:maintainer: <devops@eitr.tech>
+:maturity: new
+:depends: OpenSSL
+'''
+
+# Import Python libs
+from __future__ import absolute_import, unicode_literals
+from datetime import datetime
+import logging
+
+# Import salt libs
+# pylint: disable=import-error,no-name-in-module,redefined-builtin,3rd-party-module-not-gated
+from salt.six.moves import map as _map
+from salt.six.moves import range as _range
+# pylint: enable=import-error,no-name-in-module,redefined-builtin,3rd-party-module-not-gated
+import salt.utils.files
+
+
+# Import Third Party Libs
+try:
+    from OpenSSL import crypto
+    HAS_OPENSSL = True
+except ImportError:
+    HAS_OPENSSL = False
+
+log = logging.getLogger(__name__)
+
+DEFAULT_NOTIFY_DAYS = 45
+
+__virtualname__ = 'cert_info'
+
+
+def __virtual__():
+    if HAS_OPENSSL is False:
+        return False
+
+    return __virtualname__
+
+
+def validate(config):
+    '''
+    Validate the beacon configuration
+    '''
+    _config = {}
+    list(_map(_config.update, config))
+
+    # Configuration for cert_info beacon should be a list of dicts
+    if not isinstance(config, list):
+        return False, ('Configuration for cert_info beacon must be a list.')
+
+    if 'files' not in _config:
+        return False, ('Configuration for cert_info beacon '
+                       'must contain files option.')
+    return True, 'Valid beacon configuration'
+
+
+def beacon(config):
+    '''
+    Monitor the certificate files on the minion.
+
+    Specify a notification threshold in days and only emit a beacon if any certificates are
+    expiring within that timeframe or if `notify_days` equals `-1` (always report information).
+
+    .. code-block:: yaml
+
+        beacons:
+          cert_info:
+            - files:
+                - /etc/pki/tls/certs/mycert.pem
```

Comment:
It would be probably helpful to be able to also specify per-certificate `notify_days`, as for auto-renewing certs (e.g. provided via ACME/Let'sEncrypt) I just want to be notified if they're below their auto-renewal limit, while for all managed certificates I might want to use quite a few more days in advance...

Reply:
@eliasp check the latest commit. something like that?

Reply:
Looks great, thank you!